src/wallet/node_rpc_proxy.cpp

```diff
@@ -392,4 +392,34 @@ boost::optional<std::string> NodeRPCProxy::get_rpc_payment_info(bool mining, boo
   return boost::none;
 }
 
+boost::optional<std::string> NodeRPCProxy::get_transactions(const std::vector<crypto::hash> &txids, const std::function<void(const cryptonote::COMMAND_RPC_GET_TRANSACTIONS::request, const cryptonote::COMMAND_RPC_GET_TRANSACTIONS::response, bool)> &f)
```

Comment:
What's the point of passing the request object here when we already know what it's going to be? Also shouldn't this a `const&` not `const`? You could also speed up the response processing by passing the repsonse object by move.

Reply:
Oh well. When I saw the mail informing me about this comment I was confused why code gets commented that has not changed at all, until I saw that this is code that @j-berman added ...

We have to finish this and then merge, I tell you.

Maybe @j-berman can reply.

Reply:
> What's the point of passing the request object here when we already know what it's going to be?

`NodeRPCProxy::get_transactions` slices requests internally using the `RESTRICTED_TRANSACTIONS_COUNT`, and then this callback matches each sliced request to sliced response (so the caller can then construct the complete response how it wants to). Seemed a sensible way to structure this code, since it seems clients should always slice the request and might handle the response in a callback differently depending on context (e.g. fail early or continue).

> Also shouldn't this a const& not const? You could also speed up the response processing by passing the repsonse object by move.

Patch:

```cpp
diff --git a/src/wallet/node_rpc_proxy.cpp b/src/wallet/node_rpc_proxy.cpp
index aeb49d8e8..5980aa072 100644
--- a/src/wallet/node_rpc_proxy.cpp
+++ b/src/wallet/node_rpc_proxy.cpp
@@ -392,7 +392,7 @@ boost::optional<std::string> NodeRPCProxy::get_rpc_payment_info(bool mining, boo
   return boost::none;
 }
 
-boost::optional<std::string> NodeRPCProxy::get_transactions(const std::vector<crypto::hash> &txids, const std::function<void(const cryptonote::COMMAND_RPC_GET_TRANSACTIONS::request, const cryptonote::COMMAND_RPC_GET_TRANSACTIONS::response, bool)> &f)
+boost::optional<std::string> NodeRPCProxy::get_transactions(const std::vector<crypto::hash> &txids, const std::function<void(const cryptonote::COMMAND_RPC_GET_TRANSACTIONS::request&, const cryptonote::COMMAND_RPC_GET_TRANSACTIONS::response&, bool)> &f)
 {
   const size_t SLICE_SIZE = 100; // RESTRICTED_TRANSACTIONS_COUNT as defined in rpc/core_rpc_server.cpp
   for (size_t offset = 0; offset < txids.size(); offset += SLICE_SIZE)
@@ -417,7 +417,7 @@ boost::optional<std::string> NodeRPCProxy::get_transactions(const std::vector<cr
         check_rpc_cost(m_rpc_payment_state, "/gettransactions", resp_t.credits, pre_call_credits, resp_t.txs.size() * COST_PER_TX);
     }
 
-    f(req_t, resp_t, r);
+    f(std::move(req_t), std::move(resp_t), r);
   }
   return boost::optional<std::string>();
 }
diff --git a/src/wallet/node_rpc_proxy.h b/src/wallet/node_rpc_proxy.h
index aabf426ee..47a0ff101 100644
--- a/src/wallet/node_rpc_proxy.h
+++ b/src/wallet/node_rpc_proxy.h
@@ -59,7 +59,7 @@ public:
   boost::optional<std::string> get_dynamic_base_fee_estimate_2021_scaling(uint64_t grace_blocks, std::vector<uint64_t> &fees);
   boost::optional<std::string> get_fee_quantization_mask(uint64_t &fee_quantization_mask);
   boost::optional<std::string> get_rpc_payment_info(bool mining, bool &payment_required, uint64_t &credits, uint64_t &diff, uint64_t &credits_per_hash_found, cryptonote::blobdata &blob, uint64_t &height, uint64_t &seed_height, crypto::hash &seed_hash, crypto::hash &next_seed_hash, uint32_t &cookie);
-  boost::optional<std::string> get_transactions(const std::vector<crypto::hash> &txids, const std::function<void(const cryptonote::COMMAND_RPC_GET_TRANSACTIONS::request, const cryptonote::COMMAND_RPC_GET_TRANSACTIONS::response, bool)> &f);
+  boost::optional<std::string> get_transactions(const std::vector<crypto::hash> &txids, const std::function<void(const cryptonote::COMMAND_RPC_GET_TRANSACTIONS::request&, const cryptonote::COMMAND_RPC_GET_TRANSACTIONS::response&, bool)> &f);
 
 private:
   template<typename T> void handle_payment_changes(const T &res, std::true_type) {
```

Reply:
Patch applied, subjected to a short test, and force-pushed.